### PR TITLE
provider/google: Avoid NPE in load balancer caching agent.

### PR DESCRIPTION
**Pull request by duftler**
provider/google: Avoid NPE in load balancer controller when no matching load balancer is found.

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy [hunk 1]
```diff
@@ -234,7 +234,7 @@ class GoogleLoadBalancerCachingAgent extends AbstractGoogleCachingAgent implemen
   class HttpHealthCheckCallback<HttpHealthCheck> extends JsonBatchCallback<HttpHealthCheck> implements FailureLogger {
 
     GoogleLoadBalancer googleLoadBalancer
-    TargetPool targetPool
+    def targetPool
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy]: We were encountering what looked like this: https://jira.atlassian.com/plugins/servlet/mobile#issue/CLOV-1779/comment/778908

Cache population was failing with NPEs resulting in no populated load balancers.

**Comment by duftler** [comment]
TBR @ttomsu .

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController.groovy [hunk 2]
```diff
@@ -73,6 +73,10 @@ class GoogleLoadBalancerController {
       view.account == account && view.region == region
     }
 
+    if (!view) {
+      return []
```
Comment (ttomsu) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController.groovy]: This should probably throw a 404 to be consistent, no?

Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController.groovy]: I did it this way to be consistent with the entry-point that just takes `name`, and with the behavior of `aws/loadBalancers`.

**Pull request merged**